include/kimera_vio_ros/RosPublishers.h

```diff
@@ -0,0 +1,152 @@
+#pragma once
+
+#include <string>
+#include <unordered_map>
+
+#include <glog/logging.h>
+
+#include <image_transport/image_transport.h>
+#include <ros/ros.h>
+
+namespace VIO {
+
+/**
+* @brief The RosPublishers class
```

Comment:
Why are we making our own RosPublishers class instead of using the existing ones?

Reply:
So the problem is that you need to know before hand what outputs you want to publish for a regular ros::Publisher. In the case where you don't know how many images the VIO will want to publish nor you know what those images are, I wrap the ros::Publisher and the image_transport::Publisher so that whenever calls `publish`, it will create a ros:: or image_transport::Publisher with the name of the image (if there was no publisher like that before). It leads to code that is compact and super scalable. Checkout its usage in the code in RosDisplay.cpp:
```
    image_publishers_->publish(
        img_to_display.name_,
        cv_bridge::CvImage(header, "bgr8", img_to_display.image_).toImageMsg());
```
Above the pipeline sends an array of images to display with name and the image.
So instead of hardcoding ros::Publishers, I use the image_publishers_ class which does all the job for me.
I used this in our 3D DSG paper where I had to publish a ton of topics (one per each segmented object) without knowing beforehand how many there will be, it turned out to be extremely useful!